### PR TITLE
Edit docs to clarify role of static folder

### PR DESCRIPTION
**Pull request by codesections**
The existing docs briefly mentions the static folder under `directory structure`, by stating that the static folder can "contain[] any kind of files. All the files/folders in the static folder will be copied as-is in the output directory."

However, the role of the static folder is not mentioned when discussing how to add content to the site.  This commit adds a short section to the `Content -- overview` section explaining how to use the static folder to add static content and providing a few use cases.

(I'm suggesting this addition because I spent a while earlier today struggling to add the keybase.txt file required for [my keybase proof](https://keybase.io/codesections).  I was looking for a way to have a file in `content` that wasn't parsed as markdown, and didn't realize that I should be saving the file to `static` and skipping `content` entierly.)

### CODE REVIEW

File: docs/content/documentation/content/overview.md [hunk 1]
```diff
@@ -63,3 +63,14 @@ the public web site. You can achieve this by simply setting `ignored_content` in
...
+## Static assets
+
+In addition to placing content files in the `content` directory, you may also place content 
+files in the `static` directory.  Any files/folders that you place in the `static` directory
+will be copied, without modification, to the public directory.  
+
+Typically, you might put site-wide assets (such as the site favicon, site logos or site-wide
+JavaScript) in the root of the static directory.  You can also place any HTML or other files that
+you wish to be included without modification (that is, without being parsed as Markdown files)
+into the static directory.
```
Comment (Keats) [inline docs/content/documentation/content/overview.md]: The last sentence is not 100% true, you can also put non-md files in `content` as colocated-assets (https://www.getgutenberg.io/documentation/content/overview/#assets-colocation) as mentioned above in that file and they won't be modified either.

**Review by Keats — CHANGES_REQUESTED** [review]

File: docs/content/documentation/content/overview.md (hunk 1 above)
Comment (codesections) [reply docs/content/documentation/content/overview.md]: I added an example that clarifies that you can either colocate files or add them to the static folder, depending on which system fits your desired organazational structure.  Please let me know if you would like me to revise further.

**Comment by codesections** [comment]
Ok, revised along the lines you suggested.  I'm happy to make any further changes.

Also, my appologies for the messy commit histrory; I was trying to open a second pull request and inadvertently eddited this one and had to revert.  I don't think it will impact the master commit history, but let me know if you'd like me to rebase to clean it up.

**Comment by Keats** [comment]
Thanks!

**Pull request merged**